src/client/auth/oidc.rs

```diff
@@ -16,32 +14,26 @@ use crate::{
         },
         options::ServerApi,
     },
-    cmap::Connection,
+    cmap::{Command, Connection},
     error::{Error, Result},
     BoxFuture,
 };
+use bson::{doc, rawdoc, Document};
 
 use super::{sasl::SaslContinue, Credential, MONGODB_OIDC_STR};
 
+const HUMAN_CALLBACK_TIMEOUT: Duration = Duration::from_secs(5 * 60);
+const MACHINE_CALLBACK_TIMEOUT: Duration = Duration::from_secs(60);
+const MACHINE_INVALIDATE_SLEEP_TIMEOUT: Duration = Duration::from_millis(100);
+const API_VERSION: u32 = 1;
+
 /// The user-supplied callbacks for OIDC authentication.
 #[derive(Clone)]
 pub struct State {
     callback: Callback,
     cache: Arc<RwLock<Cache>>,
```

Comment:
Does this need to be held across `await` points?  If not, this could become a `std::sync::RwLock` and avoid the `async` that has to propagate in various places here.

Reply:
Let me check

Reply:
Looks like it will work. I didn't realize that it was safe to use std::sync::RwLock as long as it doesn't cross await, but that makes sense. I assume the only difference of the tokio version is it handles locking when the await is rescheduled.

Reply:
Yup, exactly.